### PR TITLE
feat(rest): new endpoint /components/recentComponents

### PR DESCRIPTION
**Pull request by rudra-superrr**
Issue: #1846 

Description: this endpoint returns the recent created 5 components in alphabetical order.

How to test:
Run this endpoint and you will get recent created components as response.

### CODE REVIEW

**Comment by nikkuma7** [comment]
Test successful.

File: rest/resource-server/src/docs/asciidoc/components.adoc [hunk 1]
```diff
@@ -48,6 +48,23 @@ include::{snippets}/should_document_get_components_with_fields/http-response.ado
 ===== Links
 include::{snippets}/should_document_get_components_with_fields/links.adoc[]
 
+[[resources-recent-components-list-with-fields]]
```
[the remark this answers is not in the document]
Comment (rudra-superrr) [reply rest/resource-server/src/docs/asciidoc/components.adoc]: made the changes

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/component/ComponentController.java [hunk 2]
```diff
@@ -177,6 +177,21 @@ public ResponseEntity<EntityModel<Component>> getComponent(
...
+        List<Component> sw360Components = componentService.getRecentComponents(user);
+
+        List<EntityModel> resources = new ArrayList<>();
+        sw360Components.forEach(c -> {
+            Component embeddedComponent = restControllerHelper.convertToEmbeddedComponent(c);
+            resources.add(EntityModel.of(embeddedComponent));
+        });
+
+        CollectionModel<EntityModel> finalResources = CollectionModel.of(resources);
+        return new ResponseEntity(finalResources, HttpStatus.OK);
```
[the remark this answers is not in the document]
Comment (rudra-superrr) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/component/ComponentController.java]: added <> to ResponseEntity

File: rest/resource-server/src/docs/asciidoc/components.adoc [hunk 3]
```diff
@@ -31,6 +31,25 @@ include::{snippets}/should_document_get_components/http-response.adoc[]
 ===== Links
 include::{snippets}/should_document_get_components/links.adoc[]
 
+
+[[resources-recent-components-list]]
+==== Listing recent components
+
+A `GET` request will list all of the service's recent components.
```
[the remark this answers is not in the document]
Comment (rudra-superrr) [reply rest/resource-server/src/docs/asciidoc/components.adoc]: done

**Review by afsahsyeda — APPROVED** [review]

**Pull request merged**